lib/dockerregistry/inventory.go

```diff
@@ -2158,13 +2164,19 @@ func (sc *SyncContext) Promote(
 	if customProcessRequest != nil {
 		processRequest = *customProcessRequest
 	}
-	sc.ExecRequests(populateRequests, processRequest)
+	errorTracker := false
+
+	sc.ExecRequests(populateRequests, processRequest, &errorTracker)
```

Comment:
I think you can just make ExecRequests return an error type. That way, you don't have to pass in `errorTracker` by reference to modify it. And also, you don't need to instantiate the variable outside of ExecRequests.